src/pydvl/utils/numeric.py

```diff
@@ -221,13 +241,29 @@ def linear_regression_analytical_derivative_d_x_d_theta(
     return full_derivative / N  # type: ignore
 
 
-# FIXME: FloatOrArray doesn't really work
+@overload
+def running_moments(
+    previous_avg: float, previous_variance: float, count: int, new_value: float
+) -> Tuple[float, float]:
+    ...
+
+
+@overload
+def running_moments(
+    previous_avg: "NDArray[np.float_]",
+    previous_variance: "NDArray[np.float_]",
+    count: int,
+    new_value: "NDArray[np.float_]",
+) -> Tuple["NDArray[np.float_]", "NDArray[np.float_]"]:
+    ...
+
+
 def running_moments(
-    previous_avg: FloatOrArray,
-    previous_variance: FloatOrArray,
-    count: IntOrArray,
-    new_value: FloatOrArray,
-) -> Tuple:  # [FloatOrArray, FloatOrArray]:
+    previous_avg: Union[float, "NDArray[np.float_]"],
```

Comment:
Why not keep the typevar?

Reply:
Because it would only be a shortcut for the overload, not really a type to be used elsewhere and I preferred the explicit version. The TypeVar should also be called `FloatOrArrayOfFloat` to be precise, which is not really that much shorter anyway

Reply:
I see. The typevar takes less vertical space and is less confusing so I would still prefer it but I'll leave the decision to you.

Reply:
I compromised with `__future__.annotations` and or'ing types